console/console-init/ui/src/modules/project/containers/ProjectToolbarContainer/ProjectToolbarContainer.tsx

```diff
@@ -0,0 +1,268 @@
+  const [namespaceInput, setNamespaceInput] = useState<string>();
+
+  const onClearAllFilters = () => {
+    setFilter(initialiseFilterForProject());
+  };
+
+  const onSelectDeleteAll = async (event: any) => {
+    if (event.target.value === "deleteAll") {
+      await onDeleteAll();
+    }
+  };
+
+  //this function used to clear value of type ahead select input field on filter change
+  const resettInitialState = () => {
+    setNameSelected(undefined);
+    setNameInput(undefined);
+    setNamespaceInput(undefined);
+    setNamespaceSelected(undefined);
+  };
+
+  const onFilterSelect = (value: string) => {
+    setFilter({ ...filter, filterType: value });
+    resettInitialState();
+  };
+
+  const onNameSelect = (e: any, selection: SelectOptionObject) => {
+    setNameSelected(selection.toString());
+    setNameInput(undefined);
+  };
+
+  const onNameClear = () => {
+    setNameSelected(undefined);
+    setNameInput(undefined);
+  };
+
+  const getAddressSpaceForNameORNameSpace = async (
+    value: string,
+    propertyName: string
+  ) => {
+    const response = await client.query<
+      ISearchNameOrNameSpaceAddressSpaceListResponse
+    >({
+      query: RETURN_ALL_ADDRESS_SPACES_FOR_NAME_OR_NAMESPACE(
+        propertyName,
+        value.trim()
+      ),
+      fetchPolicy: FetchPolicy.NETWORK_ONLY
+    });
+    return response;
+  };
+
+  const getOptions = (
+    response: ISearchNameOrNameSpaceAddressSpaceListResponse,
+    propertyname: string
+  ) => {
+    if (
+      response.addressSpaces &&
+      response.addressSpaces.addressSpaces &&
+      response.addressSpaces.addressSpaces.length > 0
+    ) {
+      let obtainedList = response.addressSpaces.addressSpaces.map(
+        (link: any) => link.metadata[propertyname]
+      );
+      //get list of unique records to display in the select dropdown based on total records and 100 fetched objects
+      const filteredNameOptions = getSelectOptionList(
+        obtainedList,
+        response.addressSpaces.total
+      );
+      if (filteredNameOptions.length > 0) return filteredNameOptions;
+    }
+  };
+
+  const onChangeNameInput = async (value: string) => {
+    const response = await getAddressSpaceForNameORNameSpace(value, "name");
+    return getOptions(response.data, "name");
+  };
+
+  const onNamespaceSelect = (e: any, selection: SelectOptionObject) => {
+    setNamespaceSelected(selection.toString());
+  };
+
+  const onNamespaceClear = () => {
+    setNamespaceSelected(undefined);
+    setNamespaceInput(undefined);
+  };
+
+  const onChangeNamespaceInput = async (value: string) => {
+    const response = await getAddressSpaceForNameORNameSpace(
+      value,
+      "namespace"
+    );
+    return getOptions(response.data, "namespace");
+  };
+
+  const onTypeSelect = (selection: string) => {
+    setFilter({ ...filter, type: selection });
+  };
+
+  const onSearch = () => {
+    const { filterType, names, namespaces } = filter;
+    if (filterType) {
```

Comment:
Can these checks be performed in a single if?

Reply:
No, it is not possible. there is a logic for this